compiler/src/dotty/tools/dotc/transform/TreeMapWithStages.scala

```diff
@@ -92,6 +92,13 @@ abstract class TreeMapWithStages(@constructorOnly ictx: Context) extends TreeMap
       }
 
       tree match {
+        case Apply(Select(Quoted(quotedTree), _), _) if quotedTree.isType =>
```

Comment:
The pattern does not seem to have enough guards, could it match unintended trees by accident?

Reply:
The guard are inside the `Quoted` extractor. Only the compiler generates these trees and the rest of the shape is know. I will do a refactor of the `Quoted`/`Spliced` extractor at some point, but I would prefer to do it on it's own PR.